lib/Semantics/resolve-names.cpp

```diff
@@ -3067,6 +3031,19 @@ void DeclarationVisitor::Post(const parser::EnumDef &) {
   enumerationState_ = EnumeratorState{};
 }
 
+bool DeclarationVisitor::Pre(const parser::AccessSpec &x) {
+  Attr attr{AccessSpecToAttr(x)};
+  const Scope &scope{
+      currScope().IsDerivedType() ? currScope().parent() : currScope()};
+  if (!scope.IsModule()) {
+    Say(currStmtSource().value(),
+        "%s attribute may only appear in the specification part of a module"_err_en_US,
```

Comment:
module or submodule

Reply:
That's what I thought too, because it makes perfect sense to allow it in a submodule. But:
```
C817 An access-spec shall appear only in the specification-part of a module.
C869 (R827) An access-stmt shall appear only in the specification-part of a module.
```
I'll add some comments referring to these constraints.